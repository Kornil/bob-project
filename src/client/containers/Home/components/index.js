export { default as NewUserForm } from './NewUserForm';
export { default as Card } from './Card';
