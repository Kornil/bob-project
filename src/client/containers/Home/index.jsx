import React, { Component } from 'react';
import styled from 'react-emotion';

import { NewUser, Card } from './components';

const StyledCardsContainer = styled.div`
  display: grid;
  grid-template-columns: 1fr 1fr 1fr;
  grid-gap: 8%;
`;

class Home extends Component {
  state = {
    users: [],
    status: '',
  };

  async componentDidMount() {
    await this.fetchData();
  }

  // prevent async data leaking
  componentWillUnmount() {
    this.setState = () => undefined;
  }

  fetchData = async () => {
    try {
      const response = await fetch('/get_data');
      const { users } = await response.json();
      this.setState({
        users,
        status: 'success',
      });
    } catch (err) {
      this.setState({
        status: 'error',
      });
    }
  };

  render() {
    const { users } = this.state;
    return (
      <div>
        <NewUser fetchData={this.fetchData} />
        <StyledCardsContainer>
          {users.map(user => (
            <Card key={user._id} user={user} />
          ))}
        </StyledCardsContainer>
      </div>
    );
  }
}

export default Home;
