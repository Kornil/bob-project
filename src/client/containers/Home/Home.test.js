import React from "react";
import { shallow } from "enzyme";
import Home from "./";
import MockProvider from "../../tests/mockProvider";

it("renders without crashing", () => {
  const snap = shallow(
    <MockProvider>
      <Home />
    </MockProvider>
  );
  expect(snap).toMatchSnapshot();
});
