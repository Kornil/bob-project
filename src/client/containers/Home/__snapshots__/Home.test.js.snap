// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<div>
  <h2>
    Home Page
  </h2>
  <NewUser
    fetchData={[Function]}
  />
</div>
`;
